sdk/cosmos/src/clients/collection_client.rs

```diff
@@ -60,8 +63,26 @@ impl CollectionClient {
     }
 
     /// create a document in a collection
-    pub fn create_document(&self) -> requests::CreateDocumentBuilder<'_, '_> {
-        requests::CreateDocumentBuilder::new(self)
+    pub async fn create_document<D: Serialize>(
+        &self,
+        ctx: Context,
+        document: &D,
+        options: CreateDocumentOptions<'_>,
+    ) -> Result<CreateDocumentResponse, crate::Error> {
+        let mut request = self
+            .cosmos_client()
+            .prepare_request_pipeline("collection", http::Method::POST);
+        let mut pipeline_context = PipelineContext::new(ctx, ResourceType::Collections.into());
+
+        options.decorate_request(&mut request, document)?;
```

Comment:
A couple of things to change:

1. According to [https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-document](https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-document), the URI should be `https://{databaseaccount}.documents.azure.com/dbs/{db-id}/colls/{coll-id}/docs`  (this is `https://{databaseaccount}.documents.azure.com/collection` instead).
2. The `ResourceType` is `Documents` and not `Collections` (if the resource type is wrong the authorization will fail).
3. The generated partition key is wrong. For example, running the example `document_00`, the expected header should be `"x-ms-documentdb-partitionkey": "[\"unique_id100\"]` but the generated one is `"x-ms-documentdb-partitionkey": "[{\"id\":\"unique_id100\",\"a_string\":\"Something here\",\"a_number\":100,\"a_timestamp\":1627459947}]"` which I believe is the whole document.